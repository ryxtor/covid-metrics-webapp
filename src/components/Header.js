import { NavLink } from 'react-router-dom';
import mic from '../assets/icons/microphone.png';
import settings from '../assets/icons/settings.png';

const Header = () => (
  <header>
    <NavLink className="back-home" to="/">
      &#60;
    </NavLink>
    <h1 className="header-title">Covid Stats</h1>
    <div>
      <img className="mic-img" src={mic} alt="microphone" />
      <img className="settings-img" src={settings} alt="settings" />
    </div>
  </header>
);

export default Header;
