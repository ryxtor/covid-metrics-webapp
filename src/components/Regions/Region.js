import PropTypes from 'prop-types';
import arrowLogo from '../../assets/right-arrow.png';

const Region = ({
  id, name, todayConfirmed, todayDeaths, filter, flag,
}) => {
  console.log(id);
  return (
    <div id={id} className="region-card">
      <img className="arrow-logo" src={arrowLogo} alt="arrow" />
      <img className="region-flag" src={flag.flag} alt={flag.name} />
      <div className="region-stats">
        <p className="region-name">{name}</p>
        { filter === 'default' && <p>{todayConfirmed}</p>}
        { filter === 'confirmed' && <p>{todayConfirmed}</p>}
        { filter === 'deaths' && <p>{todayDeaths}</p>}
      </div>
    </div>
  );
};

Region.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  todayConfirmed: PropTypes.number.isRequired,
  todayDeaths: PropTypes.number.isRequired,
  flag: PropTypes.instanceOf(Object).isRequired,
  filter: PropTypes.string.isRequired,
};

export default Region;
