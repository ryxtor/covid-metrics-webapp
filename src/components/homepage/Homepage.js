const Homepage = () => (
  <div>
    Under construction
  </div>
);

export default Homepage;
