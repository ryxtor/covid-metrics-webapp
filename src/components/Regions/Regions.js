import { useSelector } from 'react-redux';
import Region from './Region';

const Regions = () => {
  const regions = useSelector((state) => state.regionsReducer);

  return (
    <div>
      <p>Stats by regions</p>
      <div>
        {
          regions.regions.map((region) => (
            <Region
              data={region}
              id={region.id}
              name={region.name}
              todayConfirmed={region.todayConfirmed}
              todayDeaths={region.todayDeaths}
              key={region.id}
            />
          ))
        }
        asd
      </div>
    </div>
  );
};

export default Regions;
