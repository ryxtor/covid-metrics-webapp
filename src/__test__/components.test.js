import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';
import Homepage from '../components/homepage/Homepage';
import Regions from '../components/Regions/Regions';
import store from '../redux/configureStore';

describe('Testing Components', () => {
  test('Render Header', () => {
    const tree = renderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Homepage', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Homepage />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Regions', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Regions />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
