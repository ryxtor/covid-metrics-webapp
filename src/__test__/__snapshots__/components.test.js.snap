// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Render Header 1`] = `
<header>
  <a
    aria-current="page"
    className="back-home active"
    href="/"
    onClick={[Function]}
  >
    &lt;
  </a>
  <h1
    className="header-title"
  >
    Covid Stats
  </h1>
  <div>
    <img
      alt="microphone"
      className="mic-img"
      src="microphone.png"
    />
    <img
      alt="settings"
      className="settings-img"
      src="settings.png"
    />
  </div>
</header>
`;

exports[`Testing Components Render Homepage 1`] = `
<div
  className="homepage"
>
  <div
    className="uk-container"
  >
    <img
      alt="map"
      className="uk-map"
      src="uk-map.png"
    />
    <div
      className="uk-stats"
    >
      <p>
        UNITED KINGDOM
      </p>
      <p>
        Total Confirmed: undefined
      </p>
      <p>
        Total Deaths: undefined
      </p>
      <p>
        Today's Confirmed: undefined
      </p>
      <p>
        Today's Deaths: undefined
      </p>
    </div>
  </div>
  <div>
    <div
      className="title-filter-container"
    >
      <p
        className="region-title"
      >
        STATS BY REGIONS
      </p>
      <div>
        FILTER:
      </div>
    </div>
    <div
      className="region-cards-container"
    />
  </div>
</div>
`;

exports[`Testing Components Render Regions 1`] = `
<div>
  <div
    className="title-filter-container"
  >
    <p
      className="region-title"
    >
      STATS BY REGIONS
    </p>
    <div>
      FILTER:
    </div>
  </div>
  <div
    className="region-cards-container"
  />
</div>
`;
